pkg/cmd/infra/router/template_test.go

```diff
@@ -0,0 +1,135 @@
+			expectedValue:      nil,
+			expectErrorMessage: "failed to decode percent encoding: %25%7B%2BQ%7D%5Bssl_c_der%2Cbase64%5",
+		},
+		{
+			description:        "invalid incorrectly encoded Action when header input spec has 2 parts",
+			inputValue:         "Accept-Encoding:Delete%2",
+			expectedValue:      nil,
+			expectErrorMessage: "failed to decode percent encoding: Delete%2",
+		},
+		{
+			description:        "invalid incorrectly encoded Action when header input spec has 3 parts",
+			inputValue:         "X-Frame-Options:DENY:Delete%2",
+			expectedValue:      nil,
+			expectErrorMessage: "failed to decode percent encoding: Delete%2",
+		},
+		{
+			description:        "invalid Action when header input spec has 3 parts",
+			inputValue:         "X-Frame-Options:DENY:Bad",
+			expectedValue:      nil,
+			expectErrorMessage: "invalid action Bad",
+		},
+		{
+			description:        "invalid Action when header input spec has 2 parts",
+			inputValue:         "X-Frame-Options:Bad",
+			expectedValue:      nil,
+			expectErrorMessage: "invalid action Bad",
+		},
+		{
+			description:        "should not allow specifying headers with no action or value",
+			inputValue:         "X-Frame-Options",
+			expectErrorMessage: "invalid HTTP header input specification: X-Frame-Options",
+		},
+		{
+			description:        "should not allow blank values for headers",
+			inputValue:         "",
+			expectedValue:      nil,
+			expectErrorMessage: "encoded header string not present.",
+		},
+		{
+			description:        "should fail with incorrect header name when HTTP header input specification has 3 parts",
+			inputValue:         "X-Frame-Options**!=:DENY:Set",
+			expectedValue:      nil,
+			expectErrorMessage: "invalid HTTP header name: X-Frame-Options**!=",
+		},
+		{
+			description:        "should fail with incorrect header name when HTTP header input specification has 2 parts",
+			inputValue:         "X-Frame-Options**!=:Delete",
+			expectedValue:      nil,
+			expectErrorMessage: "invalid HTTP header name: X-Frame-Options**!=",
+		},
+		{
+			description:        "should fail when header spec consists of more than 3 parts than Name:Value:Action",
+			inputValue:         "X-Frame-Options:DENY:Set:Invalid",
+			expectedValue:      nil,
+			expectErrorMessage: "invalid HTTP header input specification: X-Frame-Options:DENY:Set:Invalid",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			switch actualValue, actualErrorMessage := parseHeadersToBeSetOrDeleted(tc.inputValue); {
+			case !cmp.Equal(actualValue, tc.expectedValue):
+				t.Errorf(" expected %s, got %s", tc.expectedValue, actualValue)
+			case tc.expectErrorMessage == "" && actualErrorMessage != nil && actualErrorMessage.Error() != "":
+				t.Fatalf("unexpected error: %v", actualErrorMessage)
+			case tc.expectErrorMessage != "" && actualErrorMessage != nil && actualErrorMessage.Error() == "":
```

Comment:
Here you should be looking for the case where actualErrorMessage is nil or blank:
```suggestion
			case tc.expectErrorMessage != "" && (actualErrorMessage == nil || actualErrorMessage.Error() == ""):
```

Reply:
actualErrorMessage == nil

then nil.Error will crash

Reply:
ok done